lib/AffineOps/AffineOps.cpp

```diff
@@ -1636,7 +1640,12 @@ LogicalResult AffineLoadOp::verify() {
   auto mapAttr = getAttrOfType<AffineMapAttr>(getMapAttrName());
   if (mapAttr) {
     AffineMap map = getAttrOfType<AffineMapAttr>(getMapAttrName()).getValue();
-    if (map.getNumResults() != getMemRefType().getRank())
+    auto rank = getMemRefType().getRank();
+    if (rank == 0) {
+      if (map.getNumResults() - 1 != rank)
```

Comment:
if (map.getNumResults() != 1) looks simpler and rank is guaranteed to be 0 due to the condition above